### PR TITLE
Share buffers and history cyclers between buffer and project panels

### PR DESCRIPTION
**Pull request by benogle**
This is an extension of #475, and makes it so the find box in both panels feels exactly the same. 

A manic gif that probably doesnt make sense:

shared-cycler

TODO

* [x] Make it work
* [x] Add the editor extension to atom-space-pen-views https://github.com/atom/atom-space-pen-views/pull/15
* [x] Specs

### CODE REVIEW

File: lib/text-editor-view.coffee [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+class TextEditorView extends View
+  # The constructor for setting up an `TextEditorView` instance.
+  constructor: (params={}) ->
+    {mini, placeholderText, attributes, editor} = params
+    attributes ?= {}
+    attributes['mini'] = mini if mini?
+    attributes['placeholder-text'] = placeholderText if placeholderText?
+
+    if editor?
+      @element = atom.views.getView(editor)
```
Comment (benogle) [inline lib/text-editor-view.coffee]: Duplicating this file is no bueno. TODO: update atom-space-pen-views to allow for the `editor` param.

File: lib/find.coffee [hunk 2]
```diff
@@ -135,10 +135,18 @@ module.exports =
   createViews: ->
     return if @findView?
 
-    history = {@findHistory, @replaceHistory, @pathsHistory}
+    findBuffer = new TextEditor(mini: true).getBuffer()
```
Comment (nathansobo) [inline lib/find.coffee]: I'm not fully understanding what's going on here... looks like you're creating these editors just to get access to their buffers? Why not just get access to the buffer of the editors you actually create?

Comment (benogle) [inline lib/find.coffee]: Yeah. I wanted them to be consistent in the way they created the editors between the views. Looks like `TextBuffer` is exported. I can just create them directly.

Comment (nathansobo) [inline lib/find.coffee]: :+1:

**Pull request merged**